[Marcus Chua] iP

marcusczh opened the pull request:
# Duke

- First
- Second

1. First
2. Second

Here is the `main` function of the Duke class
```Java
public class Duke {
  public static void main(String[] args) {
    new Duke().run();
  }
}
```

- [ ] First task
- [X] Second task

👍 

> This is a quote

Check out the [forum](https://github.com/nus-cs2103-AY2223S1/forum/issues)

**bold**
_italics_
~~strikethrough~~

Commit:
Add Gradle support

Commit:
Level 1: Greet, Echo, Exit

Commit:
Allow chatbot to store and display text entered by user

Commit:
Add functionality to allow user to mark tasks as done or not done

Commit:
Add support for tracking Todos, Events and Deadlines

Commit:
Modify txt files for automated text UI testing

Commit:
Allow chatbot to handle incorrect inputs entered by the user

Commit:
Add support for deleting tasks from the list

Commit:
Add in javadoc comments for public methods

Commit:
Add functionality to save and load tasks using local storage

Commit:
Add in support for accepting and storing dates as LocalDate objects

Commit:
Merge branch 'branch-Level-7'

Commit:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java

Commit:
Allow chatbot to list tasks occurring on a specific date

Commit:
Add support for time of tasks by using LocalDateTime

Commit:
Enumerate command types

Commit:
Refactor code to extract out closely related code as classes

Commit:
Divide classes into packages

Commit:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

Commit:
Use Gradle to build and run Duke

Commit:
Add JUnit tests

Commit:
Add JavaDoc comments to the code

Commit:
Modify code to comply with coding standard

Commit:
Add functionality to allow users to find tasks using a keyword

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/command/AddDeadLineCommand.java
#	src/main/java/duke/command/AddTodoCommand.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/DeleteCommand.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/LoadDataCommand.java
#	src/main/java/duke/command/MarkItemCommand.java
#	src/main/java/duke/command/SaveDataCommand.java
#	src/main/java/duke/command/UnMarkItemCommand.java
#	src/main/java/duke/command/WelcomeCommand.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/test/java/duke/ParserTest.java

Commit:
Merge branch 'branch-Level-9'

Commit:
Use checkStyle to detect coding style violations

Commit:
Merge branch 'branch-A-CheckStyle'

Commit:
Rename Duke to Tuna

Commit:
Implement a GUI

Commit:
Merge branch 'branch-Level-10'

Commit:
Use Java varargs for ListCommand constructor

Commit:
Merge branch 'branch-A-Varargs'

PeiYee88 commented on src/main/java/tuna/Storage.java:
```diff
@@ -0,0 +1,113 @@
+    }
+
+    /**
+     * Save the tasks in the task list into the data file located in the specified file path.
```
Verb form for javadoc; should be 'Saves'

PeiYee88 commented on src/main/java/tuna/Ui.java:
```diff
@@ -0,0 +1,181 @@
+     *
+     * @return confirmation message to indicate tasks have been saved.
+     */
+    public String fileSavedMessage() {
```
I think the method name should be an action, so can consider 'fileSaveMessage'?

PeiYee88 commented on src/main/java/tuna/command/AddEventCommand.java:
```diff
@@ -0,0 +1,50 @@
+    /** Description of the task */
+    private String taskDescription;
+    /** When the event is occurring */
+    private String at;
```
Being a bit picky here, but perhaps can change the variable name from 'at' to' 'time'? At is a bit vague.

PeiYee88 reviewed (commented):
Your code is clean and well-written. Just some minor details that violates the coding standard that need to be edited.

PeiYee88 commented on src/main/java/tuna/Tuna.java:
```diff
@@ -0,0 +1,102 @@
+package tuna;
+
+
```
additional spaces between line 2 and line 4

PeiYee88 commented on src/main/java/tuna/Ui.java:
```diff
@@ -0,0 +1,181 @@
+     * @param task task that was un-marked as done.
+     * @return confirmation message after un-marking a task as done.
+     */
+    public String unMarkTaskMessage(Task task) {
```
method name 'unmarkTaskMessage'?

PeiYee88 commented on src/main/java/tuna/command/AddDeadLineCommand.java:
```diff
@@ -0,0 +1,50 @@
+    /** Description of the task */
+    private String taskDescription;
+    /** Deadline of the task */
+    private String by;
```
can consider changing the 'by' variable name to 'deadline' to be more descriptive

PeiYee88 commented on src/main/java/tuna/command/AddEventCommand.java:
```diff
@@ -0,0 +1,50 @@
+package tuna.command;
```
I like how you put all those command in a package. WELL DONE!

PeiYee88 reviewed (commented):
Code looks good! Just some minor edits needed.

jamietan2002 commented on src/main/java/tuna/Parser.java:
```diff
@@ -0,0 +1,68 @@
+     */
+    public Command parse(String command) throws TunaException {
+        String[] inputs = command.split(" ");
+        if (inputs.length == 1) {
```
is checking for input length necessary?

jamietan2002 commented on src/main/java/tuna/Storage.java:
```diff
@@ -0,0 +1,113 @@
+                        limit = findElem(input, "/by");
+                        String task = String.join(" ", Arrays.copyOfRange(input, 2, limit));
+                        String by = String.join(" ", Arrays.copyOfRange(input, limit + 1, input.length));
+                        tasks.addDeadLine(task, by);
```
can this code be simplified since same methods are repeated for each case?

jamietan2002 commented on src/main/java/tuna/TaskList.java:
```diff
@@ -0,0 +1,159 @@
+public class TaskList {
+
+    /** List of all the tasks */
+    private ArrayList<Task> data;
```
can a better variable name be used to represent a list?

jamietan2002 commented on src/main/java/tuna/TunaException.java:
```diff
@@ -0,0 +1,16 @@
+     * @param description description for the exception.
+     */
+    public TunaException(String description) {
+        super(description);
```
good

jamietan2002 reviewed (commented):
good job!

Commit:
Package GUI related classes

The GUI related classes are difficult to differentiate from the other
classes in the package.

Package the GUI related classes separately to improve the file structure
and enhance readability.

Commit:
Allow GUI to display all text without ellipsis

The GUI will replace text with an ellipsis when there are too many
lines.

Edit the DialogBox.fxml file to allow all text to be displayed.

Commit:
Use assert to document important assumptions

Commit:
Refactor code to improve the code quality

There is code that do not follow the code quality guidelines.

Refactor the code to adhere to the code quality guidelines and improve
readability.

Commit:
Use Java Streams to replace for-loops

marcusczh committed:
Merge pull request #1 from marcusczh/branch-A-Assertions

Use assert to document important assumptions

Commit:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/tuna/Storage.java

Commit:
Merge branch 'master' into branch-A-Streams

marcusczh committed:
Merge pull request #2 from marcusczh/branch-A-CodeQuality

Refactor code to improve the code quality

marcusczh committed:
Merge pull request #3 from marcusczh/branch-A-Streams

Use Java Streams to replace for-loops

Commit:
Set up Continuous Integration

Commit:
Bug fixes

Commit:
Add sort functionality

Commit:
Merge branch 'branch-C-Sort'

Commit:
Add Javadoc for task related classes

Commit:
Improve the GUI

Commit:
Modify responses of Tuna to user commands

Commit:
Amend Javadoc comments

Commit:
Handle possible errors

Commit:
Amend text-ui-test file to match new responses

Commit:
Add more JUnit tests

Commit:
Add data file for JUnit test classes

Commit:
Revert "Add data file for JUnit test classes"

This reverts commit 4822b9e3374e99e4dcafa287308d54cd87341ab6.

Commit:
Package utility related classes

Commit:
Add title to GUI

Commit:
Create user guide

Commit:
Refine user guide

Commit:
Update user guide

Commit:
Add credits

Commit:
Add credits

Commit:
Add credits
